pkg/controller/endpoint/endpoints_controller.go

```diff
@@ -119,6 +123,13 @@ type EndpointController struct {
 	// Added as a member to the struct to allow injection for testing.
 	podsSynced cache.InformerSynced
 
+	// endpointsLister is able to list/get pods and is populated by the shared informer passed to
+	// NewEndpointController.
+	endpointsLister corelisters.EndpointsLister
+	// endpointsSynced returns true if the endpoints shared informer has been synced at least once.
+	// Added as a member to the struct to allow injection for testing.
+	endpointsSynced cache.InformerSynced
```

Comment:
Yes, please wait for endpoint cache to sync.  You don't want to make decisions on unintentionally blank data.